src/ApiClient.ts

```diff
@@ -298,6 +317,79 @@ export default class ApiClient {
     }
   }
 
+  // Take a list of queries and execute them in batches. queryOptions can be empty
+  async batchQuery(
+    queries: QueryParams[],
+    queryOptions?: QueryStreamOptions[]
+  ): Promise<messageApi.Envelope[][]> {
+    // Require length of queries and queryOptions to be the same, or queryOptions to be empty
+    if (
+      queryOptions &&
+      queryOptions.length &&
+      queryOptions.length !== queries.length
```

Comment:
Having the query be in two arrays feels a little awkward for developers to actually use.

It feels like something closer to the proto `BatchQueryRequest` would be easier to work with than having to construct the two arrays of the same length

Reply:
makes sense, I'll probably need to create a new type then? It did seem weird that our queries are split into `QueryParams` and `QueryAllOptions | QueryStreamOptions`

Reply:
The `batchQuery` signature could be:

```
type Query {
  // union of fields from QueryParams and QueryAllOptions
}

async batchQuery(queries: Query[]): Promise<messageApi.Envelope[][]>
```

@neekolas thoughts?

Reply:
This looks great.